OsmAnd/src/net/osmand/plus/views/GPXLayer.java

```diff
@@ -326,10 +327,57 @@ private void drawSelectedFilesPoints(Canvas canvas, RotatedTileBox tileBox, List
 	private void drawSelectedFilesSegments(Canvas canvas, RotatedTileBox tileBox,
 			List<SelectedGpxFile> selectedGPXFiles, DrawSettings settings) {
 		for (SelectedGpxFile g : selectedGPXFiles) {
-			List<TrkSegment> points = g.getPointsToDisplay();
-			boolean routePoints = g.isRoutePoints();
-			updatePaints(g.getColor(), routePoints, g.isShowCurrentTrack(), settings, tileBox);
-			drawSegments(canvas, tileBox, points);
+			List<TrkSegment> segments = g.getPointsToDisplay();
+			for (TrkSegment ts : segments) {
+
+				// TODO: Select/install renderables via UI or external process!  (See comments below)
+
```

Comment:
Yes, too long for final code. I just wanted to make SURE that you gent understood the capability and why.
I wanted you to experiment with adding removing renderable when you viewed/tested so that you could see how it works.
Definitely not to be in the included code.

And YES!!!  It is in the wrong place, You have perfectly understood.
There will be some way for the user via interface or some render XML to decide which ‘renderables’ they want to see.
It might be the altitude band, or the arrows, or the ants… whatever.  It needs to b added to the TrkSegment list somewhere elsewhere.
But for TESTING, and demonstration, without UI … this was a perfect place for me to put it.

MOST of these comments are for you to accept/understand my submission. The comments like this one would NOT be required in the final code.
As to putting them in the pull request, yes OK - but sometimes there are many different areas with such a big pull request, and it’s easier for me to base the notes with the part of the code like this example. It will be a bit of a balance, but I will try to adjust.

I do disagree with your “I’m against comments” philosophy.  If you understood how many HOURS I have spent trying to follow and understand code to find something I need, where simple comments would have made that process much easier. I just cannot agree that comments are a bad thing.

> > +				// TODO: Select/install renderables via UI or external process!  (See comments below)
> > +
> This is quite a long text and doesn't look to be at the right place in sense I see the mixture of what is going on in development how it could work or how it should not work. It might not be the right place to have here.
> I think if somebody will need to dive to that code he just need to understand exactly what's going on he doesn't need to understand the plans of future implementation. 
> Personally I'm against comments because I think it is always possible to refactor the code that it doesn't have comments. For example, TrkSegment has one or more 'Renderables' - it could be clear from the code itself (1 to many relations). What is Renderable is much better to describe in the Renderable class header itself. Second part Renderables are processed asynchronously is kind of obvious because we see AsyncTask.
> "Note" is a good to know for us in pull request but not in the code may be.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/osmandapp/Osmand/pull/2379/files/109ca692f100ddb59f0e538b06c32beae5910bc0#r57645577>